### PR TITLE
Fixed little endian (and scaling/offset)

### PR DESCRIPTION
**Pull request by jeankalud**
Fixed little endian handling
Fixed offset / scaling handling

Works on my Tesla DBC (that has mixed little & big endian). Please test.

### CODE REVIEW

File: common/dbc.py [hunk 1]
```diff
@@ -160,11 +163,17 @@ def decode(self, x, arr=None, debug=False):
       # big or little endian?
       #   see http://vi-firmware.openxcplatform.com/en/master/config/bit-numbering.html
       if s[3] is False:
+        endianness = "big"
```
Comment (geohot) [inline common/dbc.py]: This variable is unused

File: common/dbc.py [hunk 2]
```diff
@@ -173,12 +182,13 @@ def decode(self, x, arr=None, debug=False):
         ival -= (1<<s[2])
 
       # control the offset
-      ival = (ival + s[6])*s[5]
-      if debug:
-        print "%40s  %2d %2d  %7.2f %s" % (s[0], s[1], s[2], ival, s[-1])
+      ival = (ival * s[5]) + s[6]
```
Comment (geohot) [inline common/dbc.py]: Wow, how did this work before?

**Review by geohot — CHANGES_REQUESTED** [review]
Remove the endianness variable if unused. Then if test outputs look good I can merge.

File: common/dbc.py (hunk 2 above)
Comment (TheMutley) [inline common/dbc.py]: It worked only because almost all the signals in the Acura dbc had all their offsets at zero!

**Comment by jeankalud** [comment]
Removed the unused debug comments. Should be good now!

**Pull request merged**